packages/react-components/src/components/tooltip-component.tsx

```diff
@@ -1,121 +1,194 @@
 import * as React from 'react';
-import ReactTooltip from 'react-tooltip';
+import { createPortal } from 'react-dom';
+import { debounce } from 'lodash';
+const { useState, useEffect, useRef, useMemo } = React;
 
-type MaybePromise<T> = T | Promise<T>;
+interface TooltipProps {
+    content?: React.ReactNode;
+    visible?: boolean;
+    fadeTransition?: number;
+}
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export interface TooltipComponentState<T = any> {
-    element?: T;
-    func?: (element: T) => MaybePromise<{ [key: string]: string } | undefined>;
-    content?: string;
+interface TooltipState {
+    content: React.ReactNode;
+    visible: boolean;
+    zIndex: number;
 }
 
-export class TooltipComponent extends React.Component<unknown, TooltipComponentState> {
-    private static readonly HOURGLASS_NOT_DONE = '&#x23f3;';
+export const TooltipComponent = ({ content = '⏳', visible = false, fadeTransition = 500 }: TooltipProps) => {
```

Comment:
I get error/warning: Missing return type on function.

Reply:
Added typing for the function.